### PR TITLE
Extend the `push` command

### PR DESCRIPTION
**Pull request by flavio**
Change the push command to take two positional parameters:

  1. The policy to be pushed (source)
  2. The registry destination (destination)

This allows to push not only local files, but also policies that are
inside of the local storage.

This makes possible to push an upstream policy into a local registry.

Fixes https://github.com/kubewarden/kwctl/issues/41

### CODE REVIEW

**Comment by flavio** [comment]
The e2e tests are broken for other reasons, I'm going to fix them ASAP

File: src/main.rs [hunk 1]
```diff
@@ -62,8 +62,8 @@ async fn main() -> Result<()> {
             (@subcommand push =>
              (about: "Pushes a Kubewarden policy to an OCI registry")
              (@arg ("docker-config-json-path"): --("docker-config-json-path") +takes_value "Path to a Docker config.json-like path. Can be used to indicate registry authentication details")
-             (@arg ("policy-path"): * -p --("policy-path") +takes_value "Policy file to push")
              (@arg ("sources-path"): --("sources-path") +takes_value "YAML file holding source information (https, registry insecure hosts, custom CA's...)")
+             (@arg ("policy"): * "Policy to push. Can be the path to a local file, or a policy URI")
              (@arg ("uri"): * "Policy URI. Supported schemes: registry://")
              (@arg ("force"): -f --("force") "push also a policy that is not annotated")
```
Comment (ereslibre) [inline src/main.rs:68]: I wonder if we should keep named arguments alphabetically sorted. `--force` was already off though.

File: src/main.rs [hunk 2]
```diff
@@ -136,11 +136,24 @@ async fn main() -> Result<()> {
...
+                    .map(|u| {
+                        if u.starts_with("registry://") {
+                            String::from(u)
+                        } else {
+                            format!("registry://{}", u)
+                        }
+                    })
+                    .unwrap();
+
+                println!("Attempting to push {:?} to {:?}", wasm_path, uri);
```
Comment (ereslibre) [inline src/main.rs:152]: I'd be silent by default, and maybe use some logging library with verbosity, structured logging... I think we can start building on that story.

File: src/main.rs [hunk 3]
```diff
@@ -149,7 +162,7 @@ async fn main() -> Result<()> {
                     }
                 }
 
-                push::push(&policy, uri, docker_config, sources).await?;
+                push::push(&policy, uri.as_str(), docker_config, sources).await?;
```
Comment (ereslibre) [inline src/main.rs:165]:
```suggestion
                push::push(&policy, &uri, docker_config, sources).await?;
```

It's personal taste again, I favour `&uri`, but whatever you prefer :)

**Review by ereslibre — CHANGES_REQUESTED** [review]
Thanks, looks great. Some comments.

File: src/main.rs [hunk 4]
```diff
@@ -62,8 +62,8 @@ async fn main() -> Result<()> {
             (@subcommand push =>
              (about: "Pushes a Kubewarden policy to an OCI registry")
              (@arg ("docker-config-json-path"): --("docker-config-json-path") +takes_value "Path to a Docker config.json-like path. Can be used to indicate registry authentication details")
-             (@arg ("policy-path"): * -p --("policy-path") +takes_value "Policy file to push")
              (@arg ("sources-path"): --("sources-path") +takes_value "YAML file holding source information (https, registry insecure hosts, custom CA's...)")
+             (@arg ("policy"): * "Policy to push. Can be the path to a local file, or a policy URI")
```
[the remark this answers is not in the document]
Comment (flavio) [reply src/main.rs:66]: This is now a positional argument, like `uri`. According to my tests everything works fine

File: src/main.rs (hunk 1 above)
Comment (flavio) [reply src/main.rs:68]: I can sort the flag alphabetically

File: src/main.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (ereslibre) [reply src/main.rs:66]: Right, perfect!

**Comment by flavio** [comment]
@ereslibre: I've submitted some fixes. Once you're fine with the PR I'll squash and merge

**Review by ereslibre — APPROVED** [review]

**Pull request merged**